druid-shell/src/platform/windows/paint.rs

```diff
@@ -35,48 +35,84 @@ use winapi::um::dcommon::*;
 use winapi::um::winuser::*;
 use winapi::Interface;
 
-use direct2d;
-use direct2d::enums::{AlphaMode, RenderTargetType};
-use direct2d::render_target::{DxgiSurfaceRenderTarget, GenericRenderTarget, HwndRenderTarget};
+// old
+//use direct2d;
+//use direct2d::enums::{AlphaMode, RenderTargetType};
+//use direct2d::render_target::{DxgiSurfaceRenderTarget, GenericRenderTarget, HwndRenderTarget};
 
+// new
+use piet_common::d2d::D2DFactory;
+use winapi::um::d2d1_1::ID2D1DeviceContext;
+use wio::com::ComPtr;
+
+use crate::platform::windows::{HwndRenderTarget, DeviceContext, DxgiSurfaceRenderTarget};
+
+// end new
 use super::error::Error;
 use super::util::as_result;
 
 /// Context for painting by app into window.
 pub struct PaintCtx<'a> {
-    pub(crate) d2d_factory: &'a direct2d::Factory,
-    pub(crate) render_target: &'a mut GenericRenderTarget,
+    pub(crate) d2d_factory: &'a D2DFactory,
```

Comment:
It seems really strange to me these fields aren't getting used. I don't know what's going on.

Reply:
From what I can tell, `PaintCtx` isn't used anywhere. There's another `PaintCtx` in `druid`, but in `druid-shell` there's no `PaintCtx` impls, and no direct construction.

The `create_render_target` and `create_render_target_dxgi` fns use those fields as args, but are not methods on `PaintCtx`.

Not sure if this clarifies anything?

Reply:
Left a TODO and commented out `PaintCtx` for now.